huawei-wmi.c

```diff
@@ -922,6 +924,79 @@ static void huawei_wmi_power_unlock_exit(struct device *dev)
 		device_remove_file(dev, &dev_attr_power_unlock);
 }
 
+/*Fan speed*/
+
+static int huawei_wmi_fan_speed_get(u8 num, int *rpm)
+{
+	u8 ret[0x100] = { 0 };
+	int err;
+	
+	union hwmi_arg arg;
+	arg.cmd = FAN_SPEED_GET;
+	arg.args[2] = num;
+
+	err = huawei_wmi_cmd(arg.cmd, ret, 0x100);
+	if (err)
```

Comment:
In which cases could this happen? Do you have any examples, or rather do you need someone to test that on an unsupported laptop model?